cargo-dist/src/tasks.rs

```diff
@@ -705,7 +707,27 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         };
 
         let templates = Templates::new()?;
-        let publish_jobs = workspace_metadata.publish_jobs.clone().unwrap_or(vec![]);
+        let publish_jobs: Vec<PublishStyle>;
+        let user_publish_jobs: Vec<PublishStyle>;
+        (publish_jobs, user_publish_jobs) = workspace_metadata
+            .publish_jobs
+            .clone()
+            .unwrap_or(vec![])
+            .into_iter()
+            .partition(|s| !matches!(s, PublishStyle::User(_)));
+        let user_publish_jobs = user_publish_jobs
+            .into_iter()
+            // Remove the @ suffix for later; we only have user jobs at this
+            // point so we no longer need to distinguish
+            .map(|s| {
+                let string = s.to_string();
+                if let Some(stripped) = string.strip_prefix("./") {
+                    stripped.to_owned()
+                } else {
+                    string
+                }
```

Comment:
the ./ is stripped by this point in the code, so no need to map

Reply:
Not true, the `publish_jobs` vec we're fetching from has the original prefixed names.

Reply:
![image](https://github.com/axodotdev/cargo-dist/assets/1136864/1f845261-30d7-4465-91a7-6c04b4ce3bec)

both versions of this code were stripping it, how are we ending up with multiple copies of the strip being needed?

Reply:
Good question! I'm... not really sure? I checked and it's absolutely true that the `FromStr` is constructing the `PublishStyle` with no `./` prefix, while the type in the task *does* have it until I strip it here.